### ISSUE TITLE
feat($dataRouterRegistry): Add ability to register error views by when method

### ISSUE THREAD

**Issue by mdvorak** [mdvorak/angular-data-router#13]
Allow this

```
$dataRouterProvider.when('x.sample', {
  status: 400,
  templateUrl: '...'
});
```

This change should unify matching of status codes and media types in the registry. ATM they are mutually exclusive.

**Comment by mdvorak** [comment]
If the way of handling and registration would be unified, it might also fix #12

**Comment by mdvorak** [comment]
Closing since v1 branch will be no longer developed.
